File: KiwiJuiceEasy.java
public class KiwiJuiceEasy {
  public int[] thePouring(int[] capacities, int[] bottles, int[] fromId, int[] toId) {
    for (int i = 0; i < fromId.length; i++) {
      if (bottles[toId[i]] + bottles[fromId[i]] <= capacities[toId[i]]) {
        bottles[toId[i]] += bottles[fromId[i]];
        bottles[fromId[i]] = 0;
      } else {
        int sub = bottles[toId[i]] + bottles[fromId[i]] - capacities[toId[i]];
        bottles[toId[i]] = capacities[toId[i]];
        bottles[fromId[i]] = sub;
      }
    }
    return bottles;
  }
}

File: kiwi_juice_easy.cpp
#include <vector>
using namespace std;

class KiwiJuiceEasy
{
public:
  vector<int>
  thePouring(vector<int> capacities, vector<int> bottles, vector<int> fromId, vector<int> toId)
  {
    int m = fromId.size();
    for (int i = 0; i < m; i++)
    {
      int move = min(capacities[toId[i]] - bottles[toId[i]], bottles[fromId[i]]);
      bottles[fromId[i]] -= move;
      bottles[toId[i]] += move;
    }
    return bottles;
  }
};